src/miner.rs

```diff
@@ -66,23 +66,24 @@ impl MinerManager {
             }
         };
 
-        for c in &self.block_channels {
-            c.send(state.clone()).await.map_err(|e| e.to_string())?;
+        if !&self.block_channel.is_closed() {
+            self.block_channel.send(state.clone()).map_err(|_e| "Failed sending block to threads")?;
```

Comment:
I think if the channel is closed we want to return an error?

Reply:
I removed the `if`. Now it will return an error if the channel is closed.